src/pg_to_evalscript/javascript_processes/sort.js

```diff
@@ -0,0 +1,50 @@
+function sort(arguments) {
+  const { data, asc = true, nodata = null } = arguments;
+  let newData = [...data];
+  let arr_of_nulls = newData.filter((x) => x === null);
+
+  newData = newData.filter((x) => x !== null);
+
+  newData.sort(compareFn);
+
+  if (asc === false) {
+    newData.reverse();
+  }
+
+  if (nodata === true) {
+    newData = [...newData, ...arr_of_nulls];
+  }
+
+  if (nodata === false) {
+    newData = [...arr_of_nulls, ...newData];
+  }
+
+  return newData;
+}
+
+function compareFn(x, y) {
+  const isValidDate = function (date) {
+    return new Date(date) !== "Invalid Date" && !isNaN(new Date(date));
```

Comment:
I experimented with `new Date(date) !== "Invalid Date"` and it always returns `true`.

To make this part useful, the `dateObject.toString()` should be used - this returns `Invalid date` for the invalid dates.

Reply:
Should be fixed now